### DISCUSSION TITLE
Implements "rename" command to change app label

### DISCUSSION THREAD

**Discussion by sys-t-m** [coderGtm/yantra-app-launcher#38] [Ideas]
### Hello,

I've recently started using this launcher and am thoroughly enjoying its design and functionality. It quickly became my go-to choice for a launcher. However, I've encountered a usability challenge due to having multiple apps with identical labels, such as:

- Google Maps and Yandex Maps (both labeled "Maps")
- Google Wallet and Wallet by BudgetBakers (both labeled "Wallet")

This situation complicates quick access and alias creation for these apps. To enhance user experience and customization, I propose the implementation of a `rename` command, allowing users to change the display label of any app directly from the launcher. Additionally, a `-restore` argument could revert any changes to the app label back to its original state.

### Suggested Implementation:

1. **Rename Command:**
   - Syntax: `rename "current_app_label" "new_app_label"`
   - Example: `rename "Wallet" "Google Wallet"`
   - Upon executing the command, the user would select the specific app package to apply the new label to.

2. **Restore Command:**
   - Syntax: `rename -restore "modified_app_label"`
   - This command would restore the original label of the app that has been previously modified.

### Benefits:
- Enhances user customization and satisfaction.
- Resolves conflicts between apps with identical names, improving app selection accuracy.
- Increases efficiency in accessing and managing apps.

### Considerations:
- I understand this feature request might involve intricate changes and considerations regarding the launcher's design and functionality. As someone with limited experience in Kotlin and app development, I'm not sure about the complexity of this implementation. However, I believe this feature could significantly improve user interaction with the launcher.

Thank you for considering this suggestion. I believe it could make a great addition to the launcher, further enhancing its usability and user personalization options.

**Comment by coderGtm** [comment]
Thanks for bringing this to light. The issue is real and needs improvement. It also happens with the `call` command.

Now while thinking of solutions, although rename is 1 possible approach, I think a better approach might be possible.

One thing that comes to mind is using the app's package name instead of the label name as it guarantees uniqueness.

**Reply by sys-t-m** [reply]
Thanks for the quick response to the post!

Can you provide an example of how the user experience might change following your idea og using app's package names instead o labels?

**Reply by coderGtm** [reply]
Sure!

This new provision would allow launching apps by their package names, like

```
launch -p com.google.maps
launch -p com.yandex.maps
```

The `-p` flag will be provided to check for package names instead of labels.

The classic launch Maps` will still work but will result in a selection popup like now.

**Reply by sys-t-m** [reply]
That's a good solution and to be honest I tried to do something like that the first time I had this issue, hoping that the launcher was able to execute.

We also have to remember that:

1. Same thing must be possible with `uninstall` command
2. It's a good idea to make possible to the user to see the app's package name in a list
Example:

```
$root> list packages
// Other packages
- Maps (com.google.maps)
- Maps (com.yandex.maps)
// Other packages
```

**Reply by coderGtm** [reply]
Done 👍

Would integrate this ASAP.

**Reply by coderGtm** [reply]
Appreciate your time and efforts 🙌🏻

**Reply by sys-t-m** [reply]
Thanks to you, can't wait to see the idea implemented in the next version!

**Reply by coderGtm** [reply]
Well it won't be in the immediate next version releasing today but you can definitely expect it in the next update.

**Comment by coderGtm** [comment]
Closing this as the issue has been resolved
